typings/index.d.ts

```diff
@@ -3138,23 +3213,48 @@ declare module 'discord.js' {
 
   type MessageAdditions = MessageEmbed | MessageAttachment | (MessageEmbed | MessageAttachment)[];
 
+  interface MessageActionRowOptions extends BaseMessageComponentOptions {
+    type: 'ACTION_ROW';
+    components?: MessageComponentResolvable[];
+  }
+
   interface MessageActivity {
     partyID: string;
     type: number;
   }
 
+  interface MessageButtonOptions extends BaseMessageComponentOptions {
+    customID?: string;
+    disabled?: boolean;
+    emoji?: RawEmoji;
+    label?: string;
+    style?: MessageButtonStyle;
+    url?: string;
+  }
+
+  type MessageButtonStyle = keyof typeof MessageButtonStyles;
+
   interface MessageCollectorOptions extends CollectorOptions {
     max?: number;
     maxProcessed?: number;
   }
 
+  type MessageComponent = BaseMessageComponent | MessageActionRow | MessageButton;
+
+  type MessageComponentOptions = BaseMessageComponentOptions | MessageActionRowOptions | MessageButtonOptions;
+
+  type MessageComponentResolvable = MessageComponent | MessageComponentOptions;
+
+  type MessageComponentType = keyof typeof MessageComponentTypes;
+
   interface MessageEditOptions {
     content?: StringResolvable;
     embed?: MessageEmbed | MessageEmbedOptions | null;
     code?: string | boolean;
     flags?: BitFieldResolvable<MessageFlagsString, number>;
     allowedMentions?: MessageMentionOptions;
     attachments?: MessageAttachment[];
+    components?: BaseMessageComponent[];
```

Comment:
The code looks like it will resolve this:
```suggestion
    components?: MessageComponentResolvable[];
```